drivers:adc add max14001pmb support

NathanielAlteza opened the pull request:
Add driver implementation for MAX14001PMB.

The MAX14001 is configurable, isolated 10-bit ADCs for multi-range binary inputs.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=1718) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=1718) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/analogdevicesinc/no-OS?pullRequest=1718) it.</sub>

CiprianRegus commented:
Also, rebase on master instead of merging the master branch into yours.

NathanielAlteza commented:
Change logs:
- Renamed all "max14001pmb" to max14001.
- Put parentheses for the input (x).
- Removed the parentheses on the REVERSE_UINT16 function.
- Removed merged commit.
